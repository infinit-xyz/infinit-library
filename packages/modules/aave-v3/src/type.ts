import { Address } from 'viem'

export type AaveV3Registry = {
  poolAddressProviderRegistry?: Address
  poolAddressesProvider?: Address
  liquidationLogic?: Address
  supplyLogic?: Address
  eModeLogic?: Address
  flashLoanLogic?: Address
  borrowLogic?: Address
  bridgeLogic?: Address
  poolLogic?: Address
  configuratorLogic?: Address
  reservesSetupHelper?: Address
  poolProxy?: Address
  lendingPools?: Record<string, LendingPool>
  walletBalanceProvider?: Address
  uiIncentiveDataProvider?: Address
  uiPoolDataProviderV3?: Address
  aaveProtocolDataProvider?: Address
  l2PoolImpl?: Address
  poolConfiguratorImpl?: Address
  aclManager?: Address
  aaveOracle?: Address
  aaveEcosystemReserveV2Proxy?: Address
  aaveEcosystemReserveController?: Address
  aaveEcosystemReserveV2Impl?: Address
  reserveInterestRateStrategies?: Record<string, Address>
  wrappedTokenGatewayV3?: Address
  l2Encoder?: Address
  aTokenImpl?: Address
  delegationAwareATokenImpl?: Address
  stableDebtTokenImpl?: Address
  variableDebtTokenImpl?: Address
  poolConfiguratorProxy?: Address
  rewardsControllerImpl?: Address
  rewardsControllerProxy?: Address
  pullRewardsTransferStrategy?: Address
  emissionManager?: Address
  aggregatorApi3Adapters?: Record<string, Address>
  aggregatorPythAdapters?: Record<string, Address>
  aggregatorBandAdapters?: Record<string, Address>
}

export type LendingPool = {
  underlyingToken: Address
  interestRateStrategy: Address
  aToken: Address
  stableDebtToken: Address
  variableDebtToken: Address
}

export type Modify<T, R> = Omit<T, keyof R> & R
