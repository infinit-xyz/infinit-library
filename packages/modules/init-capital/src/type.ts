import { Address } from 'viem'

export type InitCapitalRegistry = {
  proxyAdmin?: Address
  accessControlManager?: Address
  api3ProxyOracleReaderProxy?: Address
  initOracleProxy?: Address
  configProxy?: Address
  liqIncentiveCalculatorProxy?: Address
  posManagerProxy?: Address
  initCoreProxy?: Address
  riskManagerProxy?: Address
  lsdApi3ProxyOracleReaderProxy?: Address
  pythOracleReaderProxy?: Address
  moneyMarketHookProxy?: Address
  initLens?: Address
  api3ProxyOracleReaderImpl?: Address
  initOracleImpl?: Address
  configImpl?: Address
  liqIncentiveCalculatorImpl?: Address
  posManagerImpl?: Address
  initCoreImpl?: Address
  riskManagerImpl?: Address
  lsdApi3ProxyOracleReaderImpl?: Address
  pythOracleReaderImpl?: Address
  moneyMarketHookImpl?: Address
  lendingPoolImpl?: Address
  irms?: Record<string, Address>
  lendingPools?: Record<string, LendingPools>
}

export type LendingPools = {
  underlyingToken: Address
  lendingPool: Address
  irm: Address
}
