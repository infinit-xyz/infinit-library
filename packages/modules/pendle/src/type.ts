import { Address } from 'viem'

export type PendleV3Registry = {
  pendleRouterV4?: Address
  routerStorageV4?: Address
  actionAddRemoveLiqV3?: Address
  actionCallbackV3?: Address
  actionMiscV3?: Address
  actionSimple?: Address
  actionSwapPTV3?: Address
  actionSwapYTV3?: Address
  pendleRouterStatic?: Address
  actionStorageStatic?: Address
  actionInfoStatic?: Address
  actionMarketAuxStatic?: Address
  actionMarketCoreStatic?: Address
  actionMintRedeemStatic?: Address
  actionVePendleStatic?: Address
  pendleLimitRouter?: Address
  pendleLimitRouterProxy?: Address
  proxyAdmin?: Address
  baseSplitCodeFactoryContract?: Address
  oracleLib?: Address
  pendleGaugeControllerMainchainUpg?: Address
  pendleMarketFactoryV3?: Address
  pendlePYLpOracle?: Address
  pendleSwap?: Address
  pendleYieldContractFactory?: Address
  votingEscrowPendleMainchain?: Address
  pendleMsgSendEndpointUpgImpl?: Address
  pendleMsgSendEndpointUpgProxy?: Address
  pendleVotingContollerUpgImpl?: Address
  pendleVotingControllerUpgProxy?: Address
}
