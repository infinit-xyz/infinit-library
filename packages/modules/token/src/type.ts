import { Address, Hex } from 'viem'

export enum TokenType {
  'InfinitERC20' = 'InfinitERC20',
  'InfinitERC20Burnable' = 'InfinitERC20Burnable',
}

export type Token = { type: TokenType }
export type Proxy = { proxyAddress: Address; implementation: Address }

export type TokenRegistry = {
  tokens?: Record<Address, Token>
  tokenDistributors?: Address[]
  accumulativeMerkleDistributors?: Record<Address, Proxy>
  merkleTree?: {
    root: string
    merkle: Record<Address, { amount: string; proof: Hex[] }>
  }
}
