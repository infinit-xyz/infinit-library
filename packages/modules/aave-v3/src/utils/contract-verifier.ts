import { Address, PublicClient } from 'viem'

import { BaseContractVerifier, BlockExplorerParams, ContractInfo } from '@infinit-xyz/core/internal'

import { Artifacts } from 'hardhat/types'

import { AaveV3Registry, LendingPool } from '@/src/type'
import { getArtifacts } from '@/src/utils/artifact'

export class AaveV3ContractVerifier extends BaseContractVerifier<AaveV3Registry> {
  constructor(client: PublicClient, params: BlockExplorerParams) {
    super(client, params)
  }

  protected override async getContracts(registry: AaveV3Registry): Promise<ContractInfo[]> {
    const {
      lendingPools,
      reserveInterestRateStrategies,
      poolProxy,
      poolConfiguratorProxy,
      rewardsControllerProxy,
      api3Adapters,
      pythAdapters,
      bandAdapters,
      ...contractToAddressRegistry
    } = registry

    let contracts: ContractInfo[] = []

    for (const address of Object.values(contractToAddressRegistry)) {
      contracts.push({ address })
    }

    if (registry.poolAddressesProvider) {
      if (poolProxy) {
        contracts.push({
          address: poolProxy,
          constructorArgs: [registry.poolAddressesProvider],
        })
      }
      if (poolConfiguratorProxy) {
        contracts.push({
          address: poolConfiguratorProxy,
          constructorArgs: [registry.poolAddressesProvider],
        })
      }
    }

    if (lendingPools && registry.poolConfiguratorProxy) {
      const lendingPoolContractInfos = await this.getLendingPoolContractInfos(registry.poolConfiguratorProxy, lendingPools)
      contracts = contracts.concat(lendingPoolContractInfos)
    }

    if (reserveInterestRateStrategies) {
      const reserveInterestRateStrategyContractInfos = await this.getReserveInterestRateStrategyContractInfos(reserveInterestRateStrategies)
      contracts = contracts.concat(reserveInterestRateStrategyContractInfos)
    }

    if (api3Adapters) {
      const api3AdaptersContractInfos = await this.getAdapterContractInfos(api3Adapters)
      contracts = contracts.concat(api3AdaptersContractInfos)
    }

    if (pythAdapters) {
      const pythAdaptersContractInfos = await this.getAdapterContractInfos(pythAdapters)
      contracts = contracts.concat(pythAdaptersContractInfos)
    }

    if (bandAdapters) {
      const bandAdaptersContractInfos = await this.getAdapterContractInfos(bandAdapters)
      contracts = contracts.concat(bandAdaptersContractInfos)
    }

    return contracts
  }

  protected override async getArtifacts(): Promise<Artifacts> {
    const artifacts = await getArtifacts()

    return artifacts
  }

  private async getLendingPoolContractInfos(poolConfiguratorProxy: Address, lendingPools: Record<string, LendingPool>) {
    const contracts: ContractInfo[] = []
    for (const info of Object.values(lendingPools)) {
      for (const [contractName, address] of Object.entries(info)) {
        if (contractName === 'underlyingToken') continue
        contracts.push({
          address,
          constructorArgs: contractName !== 'interestRateStrategy' ? [poolConfiguratorProxy] : undefined,
        })
      }
    }
    return contracts
  }

  private async getReserveInterestRateStrategyContractInfos(reserveInterestRateStrategies: Record<string, Address>) {
    const contracts: ContractInfo[] = []
    for (const address of Object.values(reserveInterestRateStrategies)) {
      contracts.push({ address })
    }
    return contracts
  }

  private async getAdapterContractInfos(adapters: Record<string, Address>) {
    const contracts: ContractInfo[] = []
    for (const address of Object.values(adapters)) {
      contracts.push({ address })
    }
    return contracts
  }
}
