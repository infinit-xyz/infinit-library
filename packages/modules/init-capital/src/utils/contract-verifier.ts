import { Address, PublicClient } from 'viem'

import { BaseContractVerifier, BlockExplorerParams, ContractInfo } from '@infinit-xyz/core/internal'

import { Artifacts } from 'hardhat/types'

import { InitCapitalRegistry } from '@/src/type'
import { getArtifacts } from '@/src/utils/artifact'

export class TokenContractVerifier extends BaseContractVerifier<InitCapitalRegistry> {
  constructor(client: PublicClient, params: BlockExplorerParams) {
    super(client, params)
  }

  protected override async getContracts(registry: InitCapitalRegistry): Promise<ContractInfo[]> {
    const { lendingPools, irms, ...contractToAddressRegistry } = registry
    const contracts: ContractInfo[] = []

    for (const address of Object.values(contractToAddressRegistry)) {
      contracts.push({ address })
    }

    if (lendingPools) {
      for (const { lendingPool, irm } of Object.values(lendingPools)) {
        contracts.push({ address: lendingPool })
        contracts.push({ address: irm })
      }
    }

    if (irms) {
      for (const address of Object.values(irms)) {
        contracts.push({ address })
      }
    }

    return contracts
  }

  protected override async getArtifacts(): Promise<Artifacts> {
    const artifacts = await getArtifacts()

    return artifacts
  }
}
