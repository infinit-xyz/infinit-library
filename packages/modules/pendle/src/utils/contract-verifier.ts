import { Address, PublicClient } from 'viem'

import { BaseContractVerifier, BlockExplorerParams, ContractInfo } from '@infinit-xyz/core/internal'

import { Artifacts } from 'hardhat/types'

import { PendleRegistry } from '@/src/type'
import { getArtifacts, readArtifact } from '@/src/utils/artifact'

export class PendleContractVerifier extends BaseContractVerifier<PendleRegistry> {
  constructor(client: PublicClient, params: BlockExplorerParams) {
    super(client, params)
  }

  protected override async getContracts(registry: PendleRegistry): Promise<ContractInfo[]> {
    const contracts: ContractInfo[] = []

    const { pendleMarketFactoryV3, pendleYieldContractFactory, ...contractToAddressRegistry } = registry

    for (const address of Object.values(contractToAddressRegistry)) {
      contracts.push({ address })
    }

    if (pendleYieldContractFactory) {
      const pendleYieldContractFactoryArtifact = await readArtifact('PendleYieldContractFactory')
      const ytCreationCodeContractA: Address = await this.client.readContract({
        address: registry.pendleYieldContractFactory!,
        abi: pendleYieldContractFactoryArtifact.abi,
        functionName: 'ytCreationCodeContractA',
        args: [],
      })
      const ytCreationCodeSizeA: bigint = await this.client.readContract({
        address: registry.pendleYieldContractFactory!,
        abi: pendleYieldContractFactoryArtifact.abi,
        functionName: 'ytCreationCodeSizeA',
        args: [],
      })

      const ytCreationCodeContractB: Address = await this.client.readContract({
        address: registry.pendleYieldContractFactory!,
        abi: pendleYieldContractFactoryArtifact.abi,
        functionName: 'ytCreationCodeContractB',
        args: [],
      })
      const ytCreationCodeSizeB: bigint = await this.client.readContract({
        address: registry.pendleYieldContractFactory!,
        abi: pendleYieldContractFactoryArtifact.abi,
        functionName: 'ytCreationCodeSizeB',
        args: [],
      })

      contracts.push({
        address: pendleYieldContractFactory,
        constructorArgs: [ytCreationCodeContractA, ytCreationCodeSizeA, ytCreationCodeContractB, ytCreationCodeSizeB],
      })
    }

    if (pendleMarketFactoryV3) {
      const pendleMarketFactoryV3Artifact = await readArtifact('PendleMarketFactoryV3')
      const yieldContractFactory: Address = await this.client.readContract({
        address: registry.pendleMarketFactoryV3!,
        abi: pendleMarketFactoryV3Artifact.abi,
        functionName: 'yieldContractFactory',
        args: [],
      })
      const marketCreationCodeContractA: Address = await this.client.readContract({
        address: registry.pendleMarketFactoryV3!,
        abi: pendleMarketFactoryV3Artifact.abi,
        functionName: 'marketCreationCodeContractA',
        args: [],
      })
      const marketCreationCodeSizeA: bigint = await this.client.readContract({
        address: registry.pendleMarketFactoryV3!,
        abi: pendleMarketFactoryV3Artifact.abi,
        functionName: 'marketCreationCodeSizeA',
        args: [],
      })
      const marketCreationCodeContractB: Address = await this.client.readContract({
        address: registry.pendleMarketFactoryV3!,
        abi: pendleMarketFactoryV3Artifact.abi,
        functionName: 'marketCreationCodeContractB',
        args: [],
      })
      const marketCreationCodeSizeB: bigint = await this.client.readContract({
        address: registry.pendleMarketFactoryV3!,
        abi: pendleMarketFactoryV3Artifact.abi,
        functionName: 'marketCreationCodeSizeB',
        args: [],
      })
      const treasury: Address = await this.client.readContract({
        address: registry.pendleMarketFactoryV3!,
        abi: pendleMarketFactoryV3Artifact.abi,
        functionName: 'treasury',
        args: [],
      })
      const reserveFeePercent: number = await this.client.readContract({
        address: registry.pendleMarketFactoryV3!,
        abi: pendleMarketFactoryV3Artifact.abi,
        functionName: 'reserveFeePercent',
        args: [],
      })
      const vePendle: Address = await this.client.readContract({
        address: registry.pendleMarketFactoryV3!,
        abi: pendleMarketFactoryV3Artifact.abi,
        functionName: 'vePendle',
        args: [],
      })
      contracts.push({
        address: pendleMarketFactoryV3,
        constructorArgs: [
          yieldContractFactory,
          marketCreationCodeContractA,
          marketCreationCodeSizeA,
          marketCreationCodeContractB,
          marketCreationCodeSizeB,
          treasury,
          reserveFeePercent,
          vePendle,
          registry.pendleGaugeControllerMainchainUpgProxy!,
        ],
      })
    }

    return contracts
  }

  protected override async getArtifacts(): Promise<Artifacts> {
    const artifacts = await getArtifacts()

    return artifacts
  }
}
