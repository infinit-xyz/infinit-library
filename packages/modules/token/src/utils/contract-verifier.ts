import { Address, PublicClient } from 'viem'

import { BaseContractVerifier, BlockExplorerParams, ContractInfo } from '@infinit-xyz/core/internal'

import { Artifacts } from 'hardhat/types'

import { TokenRegistry } from '@/src/type'
import { getArtifacts } from '@/src/utils/artifact'

export class TokenContractVerifier extends BaseContractVerifier<TokenRegistry> {
  constructor(client: PublicClient, params: BlockExplorerParams) {
    super(client, params)
  }

  protected override async getContracts(registry: TokenRegistry): Promise<ContractInfo[]> {
    const { accumulativeMerkleDistributors, tokens } = registry
    const contracts: ContractInfo[] = []

    if (tokens) {
      for (const address of Object.keys(tokens)) {
        contracts.push({ address: address as Address })
      }
    }
    if (accumulativeMerkleDistributors) {
      for (const [proxyAddress, { implementation }] of Object.entries(accumulativeMerkleDistributors)) {
        contracts.push({ address: proxyAddress as Address })
        contracts.push({ address: implementation })
      }
    }

    return contracts
  }

  protected override async getArtifacts(): Promise<Artifacts> {
    const artifacts = await getArtifacts()

    return artifacts
  }
}
