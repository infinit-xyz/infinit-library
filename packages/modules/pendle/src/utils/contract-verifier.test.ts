import { describe, expect, test } from 'vitest'

import { createPublicClient, http } from 'viem'
import { holesky } from 'viem/chains'

import { PendleRegistry } from '../type'
import { PendleContractVerifier } from './contract-verifier'

// Create a subclass to expose the protected method for testing
class TestPendleContractVerifier extends PendleContractVerifier {
  public async testGetContracts(registry: PendleRegistry) {
    return this.getContracts(registry)
  }
}

describe('TokenContractVerifier.ts', () => {
  const registry: PendleRegistry = {
    baseSplitCodeFactoryContract: '0xa1c1c280822a70e414bf69ba137f6357b44f52a2',
    pendleSwap: '0xc3d11902938ba2fd41b78328c0422f7eecc95cab',
    pendleMsgSendEndpointUpgImpl: '0xc34980bebe8502664ccf7d737d0a0a9b449df27d',
    pendleMsgSendEndpointUpgProxy: '0x06d46b95bcd056ba2f71c7b5c5d59518b2e5261e',
    votingEscrowPendleMainchain: '0xba76b089119c9b9a1cfebbfbb6f9af1517f492ab',
    pendleYieldContractFactory: '0x7c4ffd0cbcc09adcb44ad742af3e17a19e3a2725',
    oracleLib: '0xb9404bf03c9470aefa6f76d2d6143569487201cd',
    pendleVotingControllerUpgImpl: '0x52746b8730db3f563fa247ad1ae8107c1c55fbbe',
    pendleVotingControllerUpgProxy: '0xf498122e6ad20478e2f7f43a4f71c1f14fa913d5',
    pendleGaugeControllerMainchainUpgImpl: '0x906c6be1476d5ebe1131f90bdc88aae7fe512a0f',
    pendleGaugeControllerMainchainUpgProxy: '0xbd12bd428660046f56301ddc4c0cdeca9b53e742',
    pendleMarketFactoryV3: '0x3885c047f7a5b4266aa4f7dc55694494eec4845b',
    routerStorageV4: '0xdc4ee48d889cb5b578cda7698bdc5f9f2266a9b8',
    actionAddRemoveLiqV3: '0xed208208c60e27562cbee83d75cafdfb3feaefda',
    actionCallbackV3: '0x57f575e1714e136ae10a5cec2845e1330224a918',
    actionMiscV3: '0x7b97f317de3ecf0a91cad83b4700edd0cc4d1f89',
    actionSimple: '0x2c04da4d8606494da2efb8c1cbb4ef729b5162e2',
    actionSwapPTV3: '0x25f41915852b41782c5fcd15f0d63924bf50a752',
    actionSwapYTV3: '0x6b939c4ccb22526a1989df2dcea49d36fa097de2',
    pendleRouterV4: '0xd661062002d9319d2fdabe21d9e0fbe9ba4d5ba6',
    actionStorageStatic: '0xc61802f976c88307291b494c3105614c3c073960',
    actionInfoStatic: '0xf6051261d4e5fad492a4501c04174881c226166c',
    actionMarketAuxStatic: '0x8c2f12783b9adf4affec321e488ccd1d40298c8d',
    actionMarketCoreStatic: '0x12b83df96e6e2dd315e3fbad3439fd1aad34bfc6',
    actionMintRedeemStatic: '0x26c58376f164ff680d370f969d7f701d233d699f',
    actionVePendleStatic: '0x26ddc808243300f1285528e356f02b07d8f4420d',
    pendleRouterStatic: '0xc9569e07bb5e4e8ac6cca49d7b520b98532066ad',
    proxyAdmin: '0x595225975dde6f422328617c4c55b14010470991',
    pendleLimitRouterImpl: '0xd6dd32a87bca870819d0bfea12058ccb2d3d0dc8',
    pendleLimitRouterProxy: '0x89fa9681fb808f02b47235924f503937486ee280',
    pendlePYLpOracle: '0xd3b31da7317cd1cf8fa1398d485061513515e4ef',
    pendlePYLpOracleProxy: '0xb71309e4d3e653c5830a2cf2af5a0a8ad41488b0',
    pendleMulticallV2: '0xd45d1550b1b3e730fdf37a43cd963614513616c1',
    multicall: '0x3059dc23ca80f575d518d13276b20b6cd71734ea',
    simulateHelper: '0x36e3ee95a0e62911d40f7cf058c378d75c94d31e',
    supplyCapReader: '0x27e0f7a59eca46dff3b2a9de07385e237867df65',
    pendlePoolDeployHelper: '0x05e0a84b5effa7bf2350f3cd14633c8df6544339',
    pendleGovernanceProxyImpl: '0x214c233fa5bd380c8e9602d7b7dfb9a3d3e0f07f',
    pendleGovernanceProxy: '0x9af271d3cc828c87d2b0f6b1eb3943e2f4e0a8fd',
    pendleBoringOneracle: '0xda27654ccbb33ac3b7253add434afec793a8b4ab',
  }

  test('getContracts: should match snapshot', async () => {
    const verifier = new TestPendleContractVerifier(
      createPublicClient({
        chain: holesky,
        transport: http('https://holesky.drpc.org/'),
      }),
      {
        apiKey: process.env.HOLESKY_ETHERSCAN_API_KEY ?? '',
        apiUrl: 'https://api-holesky.etherscan.io/api',
        url: 'https://holesky.etherscan.io',
      },
    )
    expect(verifier.testGetContracts(registry)).resolves.toMatchSnapshot()
  })
})
