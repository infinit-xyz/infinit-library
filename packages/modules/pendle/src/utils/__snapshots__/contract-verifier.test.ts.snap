// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TokenContractVerifier.ts > getContracts: should match snapshot 1`] = `
[
  {
    "address": "0xa1c1c280822a70e414bf69ba137f6357b44f52a2",
  },
  {
    "address": "0xc3d11902938ba2fd41b78328c0422f7eecc95cab",
  },
  {
    "address": "0xc34980bebe8502664ccf7d737d0a0a9b449df27d",
  },
  {
    "address": "0x06d46b95bcd056ba2f71c7b5c5d59518b2e5261e",
  },
  {
    "address": "0xba76b089119c9b9a1cfebbfbb6f9af1517f492ab",
  },
  {
    "address": "0xb9404bf03c9470aefa6f76d2d6143569487201cd",
  },
  {
    "address": "0x52746b8730db3f563fa247ad1ae8107c1c55fbbe",
  },
  {
    "address": "0xf498122e6ad20478e2f7f43a4f71c1f14fa913d5",
  },
  {
    "address": "0x906c6be1476d5ebe1131f90bdc88aae7fe512a0f",
  },
  {
    "address": "0xbd12bd428660046f56301ddc4c0cdeca9b53e742",
  },
  {
    "address": "0xdc4ee48d889cb5b578cda7698bdc5f9f2266a9b8",
  },
  {
    "address": "0xed208208c60e27562cbee83d75cafdfb3feaefda",
  },
  {
    "address": "0x57f575e1714e136ae10a5cec2845e1330224a918",
  },
  {
    "address": "0x7b97f317de3ecf0a91cad83b4700edd0cc4d1f89",
  },
  {
    "address": "0x2c04da4d8606494da2efb8c1cbb4ef729b5162e2",
  },
  {
    "address": "0x25f41915852b41782c5fcd15f0d63924bf50a752",
  },
  {
    "address": "0x6b939c4ccb22526a1989df2dcea49d36fa097de2",
  },
  {
    "address": "0xd661062002d9319d2fdabe21d9e0fbe9ba4d5ba6",
  },
  {
    "address": "0xc61802f976c88307291b494c3105614c3c073960",
  },
  {
    "address": "0xf6051261d4e5fad492a4501c04174881c226166c",
  },
  {
    "address": "0x8c2f12783b9adf4affec321e488ccd1d40298c8d",
  },
  {
    "address": "0x12b83df96e6e2dd315e3fbad3439fd1aad34bfc6",
  },
  {
    "address": "0x26c58376f164ff680d370f969d7f701d233d699f",
  },
  {
    "address": "0x26ddc808243300f1285528e356f02b07d8f4420d",
  },
  {
    "address": "0xc9569e07bb5e4e8ac6cca49d7b520b98532066ad",
  },
  {
    "address": "0x595225975dde6f422328617c4c55b14010470991",
  },
  {
    "address": "0xd6dd32a87bca870819d0bfea12058ccb2d3d0dc8",
  },
  {
    "address": "0x89fa9681fb808f02b47235924f503937486ee280",
  },
  {
    "address": "0xd3b31da7317cd1cf8fa1398d485061513515e4ef",
  },
  {
    "address": "0xb71309e4d3e653c5830a2cf2af5a0a8ad41488b0",
  },
  {
    "address": "0xd45d1550b1b3e730fdf37a43cd963614513616c1",
  },
  {
    "address": "0x3059dc23ca80f575d518d13276b20b6cd71734ea",
  },
  {
    "address": "0x36e3ee95a0e62911d40f7cf058c378d75c94d31e",
  },
  {
    "address": "0x27e0f7a59eca46dff3b2a9de07385e237867df65",
  },
  {
    "address": "0x05e0a84b5effa7bf2350f3cd14633c8df6544339",
  },
  {
    "address": "0x214c233fa5bd380c8e9602d7b7dfb9a3d3e0f07f",
  },
  {
    "address": "0x9af271d3cc828c87d2b0f6b1eb3943e2f4e0a8fd",
  },
  {
    "address": "0xda27654ccbb33ac3b7253add434afec793a8b4ab",
  },
  {
    "address": "0x7c4ffd0cbcc09adcb44ad742af3e17a19e3a2725",
    "constructorArgs": [
      "0x23027b0220380615480f2E219E65fc75082cde64",
      8996n,
      "0xDFbdF029b9B07D1C6cD3eb38ffC80931E424A18a",
      8996n,
    ],
  },
  {
    "address": "0x3885c047f7a5b4266aa4f7dc55694494eec4845b",
    "constructorArgs": [
      "0x7C4FFd0cBCC09ADCb44AD742af3e17A19E3a2725",
      "0xa40337246D3BD5C75D882b41b9E3655dFDe63082",
      13398n,
      "0x597882171Dd9125E8A9846479F3D8e193d02c6EA",
      13398n,
      "0xF2c1010CF62eA54b64B3461CF7F042Cc45D9e9c7",
      10,
      "0xBa76b089119C9b9a1cfeBbFBb6F9Af1517F492aB",
      "0xbd12bd428660046f56301ddc4c0cdeca9b53e742",
    ],
  },
]
`;
